### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by nanfuka**
What does this PR do?
-it creates API endpoints for the following functions:
-Place a new order for food.
-Get a list of orders.
-Fetch a specific order.
-Update the order status.

Description of the task to be completed?
 create API endpoints for the following functions:
-Place a new order for food.
-Get a list of orders.
-Fetch a specific order.
-Update the order status.

How should this be manually tested?
The description regarding how this is to be tasted is explained in the readme file

Relevant pivotal tracker stories?
-#160751579
-#160751448
#160751579
#160751794
#160752025
#160752190
getorders-suceess
get-specific-order-fail
getorders-no-auth
get-specific-order-success
login-fail
login-success
modifyrequest-success
postorders-fail
postorders-suceess

-

### CODE REVIEW

File: api/model/User.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+class User:
+    def __init__(self, firstName, lastName, email, username, password, is_admin=False):
+        self.firstName = firstName
+        self.lastName = lastName
+        self.email = email
+        self.username = username
+        self.password = self.hash_password(password)
+        self.is_admin = is_admin
+
+    def getFirstName(self):
```
Comment (Sebuliba-Adrian) [inline api/model/User.py]: @nanfuka Good work with the OOP. But unlike languages like Java, Its best to use snake case not camel cases in python when naming your methods, functions or variables for example here you could have named this like: get_first_name(self): and for the variable first_name

File: api/model/UserRequest.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+class UserRequest:
+	def __init__(self,requestTitle,requestType,requestCategory,requestStatus,username,requestId= str(uuid.uuid4())):
+		self.requestId=requestId
+		self.requestTitle=requestTitle
+		self.requestType=requestType
+		self.requestCategory=requestCategory
+		self.requestStatus=requestStatus
+		self.owner=username
+
+	def getDictionary(self):
```
Comment (Sebuliba-Adrian) [inline api/model/UserRequest.py]: Same here use snake_case for both variables and methods

File: api/model/data.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        """
+        function to search a list of users for a specific username
+        """
+        for item in self.users:
+            if item.getUserName() == username:
+                return item
+            else:
+                return None
+
+    def generate_auth_token(self, user):
```
Comment (Sebuliba-Adrian) [inline api/model/data.py:80]: @nanfuka Yes. Rename your method like the format you used here

File: api/tests/test_TestCreateRequest.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+from app import app
+import unittest
+import json
+from api.tests.Basetest import BaseTest
+
+
+
+class TestCreateRequests(BaseTest):
+
+    # Check if URL path exists and is protected
```
Comment (Sebuliba-Adrian) [inline api/tests/test_TestCreateRequest.py]: @nanfuka  Use docstrings when commenting on your code.

File: api/tests/test_TestOrders.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+from flask_testing import TestCase
+#from app.models.UserRequest import UserRequest
```
Comment (Sebuliba-Adrian) [inline api/tests/test_TestOrders.py]: Avoid commented out code to keep your code clean.

File: venv/Include/Python.h [hunk 6]
```diff
@@ -0,0 +1,139 @@
+#ifndef Py_PYTHON_H
```
Comment (Sebuliba-Adrian) [inline venv/Include/Python.h:1]: @nanfuka I see you pushed ```venv``` folder. Please avoid doing so by  blacklisting it in your ```.gitignore``` file. Remember to reach out for help if you do not know how to do this.

**Comment by Sebuliba-Adrian** [comment]
@nanfuka  You should also use pylint/PEP8 to cleanup your code.

**Comment by aayebare** [comment]
nice PR ..great documentation

**Comment by billkabanga** [comment]
I like the fact that you are following the recommended PR syntax.
I think @Sebuliba-Adrian has pointed out most of the issues.

**Comment by kdaglas** [comment]
I took commend you for the good PR syntax though leave the hashtag before every heading so that you maintain that bold nature of the title for example ###heading

File: api/model/UserRequest.py (hunk 2 above)
Comment (nanfuka) [reply api/model/UserRequest.py]: comment seen  thanks for the feedback

**Pull request merged**